src/gmp-test/DestinationChainSwapExecutableForetellable.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { IAxelarExecutableForetellable } from '../interfaces/IAxelarExecutableForetellable.sol';
+import { IERC20 } from '../interfaces/IERC20.sol';
+import { DestinationChainTokenSwapper } from './DestinationChainTokenSwapper.sol';
+
+contract DestinationChainSwapExecutableForetellable is IAxelarExecutableForetellable {
+    DestinationChainTokenSwapper swapper;
+
+    constructor(address gatewayAddress, address swapperAddress) IAxelarExecutableForetellable(gatewayAddress) {
+        swapper = DestinationChainTokenSwapper(swapperAddress);
+    }
+
+    function _executeWithToken(
+        string memory sourceChain,
+        string memory,
+        bytes calldata payload,
+        string memory tokenSymbolA,
+        uint256 amount
+    ) internal override {
+        (string memory tokenSymbolB, string memory recipient) = abi.decode(payload, (string, string));
```

Comment:
```suggestion
        (bytes32 nonce, string memory tokenSymbolB, string memory recipient) = abi.decode(payload, (string, string));
```
Without a nonce, only one of the multiple swapping attempts with the exact same parameters can be foretold until executed.